faiss/python/__init__.py

```diff
@@ -656,23 +657,61 @@ def knn_gpu(res, xq, xb, k, D=None, I=None, metric=METRIC_L2):
 # numpy array / std::vector conversions
 ###########################################
 
-# mapping from vector names in swigfaiss.swig and the numpy dtype names
-vector_name_map = {
-    'Float': 'float32',
-    'Byte': 'uint8',
-    'Char': 'int8',
-    'Uint64': 'uint64',
-    'LongLong': 'int64',
-    'Int': 'int32',
-    'Double': 'float64'
+sizeof_long = array.array('l').itemsize
+deprecated_name_map = {
+    # deprecated: replacement
+    'Float': 'Float32',
+    'Double': 'Float64',
+    'Char': 'Int8',
+    'Int': 'Int32',
+    'Long': 'Int32' if sizeof_long == 4 else 'Int64',
+    'LongLong': 'Int64',
+    'Byte': 'UInt8',
+    # previously misspelled variant
+    'Uint64': 'UInt64',
 }
```

Comment:
_Not_ deprecating some of these (e.g. if it's considered desirable to keep `DoubleVector`) is easy: we would just need to move the line down from this dict to `vector_name_map` (and lower-case the value of the KV-pair)